zerver/lib/push_notifications.py

```diff
@@ -65,19 +55,25 @@ def __init__(self, user, tokens, alert=None, badge=None, sound=None,
         payload = Payload(alert=alert, badge=badge, sound=sound,
                           category=category, custom=kwargs)
         for token in tokens:
-            identifier = random.getrandbits(32)
-            identifier_token_map[identifier] = {'token': token, 'user': user}
+            identifier = generate_random_token(32)
+            data = {'token': token, 'user': user}
+            redis_client.hmset(identifier, data)
+            redis_client.expire(identifier, expiry)
```

Comment:
We should probably namespace the redis keys so we don't need to worry about collisions with other things we're using redis for.  E.g. keys could be `apns:randombits` rather than just `randombits`